crates/bevy_ecs/src/component.rs

```diff
@@ -138,6 +138,17 @@ impl ComponentInfo {
     }
 }
 
+/// A [`ComponentId`] is an opaque value which uniquely identifies the type of
```

Comment:
I've just realized that I write taht `ComponentId` is opaque here, but the `ComponentId::new` function is actually public. Are there any legitimate use cases in which a `bevy_ecs` consumer would need to create a `ComponentId` from scratch?

Reply:
Yes: for non-Rust types.